### PR TITLE
Prepare gdbserver for multicore support

### PR DESCRIPTION
**Pull request by SimonWin**

### CODE REVIEW

File: include/vcml/debugging/gdbserver.h [hunk 1]
```diff
@@ -35,10 +35,19 @@ enum gdb_signal {
     GDBSIG_KILL = 9,
 };
 
+struct gdb_target {
```
Comment (janweinstock) [inline include/vcml/debugging/gdbserver.h:38]: Can we move `gdb_target` inside `gdbserver`?

File: include/vcml/debugging/gdbserver.h [hunk 2]
```diff
@@ -35,10 +35,19 @@ enum gdb_signal {
...
+    u64 tid;
+    u64 pid;
+    string xml;
+    vector<const cpureg*> cpuregs;
+    target* tgt;
```
Comment (janweinstock) [inline include/vcml/debugging/gdbserver.h:43]: Could this be a reference? i.e. `target& tgt`?

File: include/vcml/debugging/gdbserver.h [hunk 3]
```diff
@@ -35,10 +35,19 @@ enum gdb_signal {
...
+};
+
 class gdbserver : public rspserver, private subscriber, private suspender
 {
 private:
-    target& m_target;
+    vector<gdb_target*> m_targets;
```
Comment (janweinstock) [inline include/vcml/debugging/gdbserver.h:49]: Can this be `vector<gdb_target>`? Or why is this a pointer?

File: src/vcml/debugging/gdbserver.cpp [hunk 4]
```diff
@@ -567,5 +573,13 @@ void gdbserver::handle_disconnect() {
         update_status(m_default);
 }
 
+void gdbserver::add_target(target* tgt) {
+    auto gtgt = new gdb_target{ .tid = tgt->core_id() + 1,
+                                .pid = 1,
+                                .xml = "",
+                                .tgt = tgt };
+    m_targets.push_back(gtgt);
```
Comment (janweinstock) [inline src/vcml/debugging/gdbserver.cpp:581]: `m_target.emplace_back(tgt->core_id() + 1, 1, "", *tgt);`

**Review by janweinstock — CHANGES_REQUESTED** [review]

File: include/vcml/debugging/gdbserver.h (hunk 1 above)
Comment (SimonWin) [reply include/vcml/debugging/gdbserver.h]: done

File: include/vcml/debugging/gdbserver.h (hunk 2 above)
Comment (SimonWin) [reply include/vcml/debugging/gdbserver.h]: done

File: include/vcml/debugging/gdbserver.h (hunk 3 above)
Comment (SimonWin) [reply include/vcml/debugging/gdbserver.h]: done

File: src/vcml/debugging/gdbserver.cpp (hunk 4 above)
Comment (SimonWin) [reply src/vcml/debugging/gdbserver.cpp]: done

File: include/vcml/debugging/gdbserver.h [hunk 5]
```diff
@@ -35,18 +35,21 @@ enum gdb_signal {
...
 private:
-    vector<gdb_target*> m_targets;
+    struct gdb_target {
+        u64 tid;
+        u64 pid;
+        string xml;
+        vector<const cpureg*> cpuregs;
+        target& tgt;
+
+        gdb_target(u64 t, u64 p, const char* x, target& tg):
```
Comment (janweinstock) [inline include/vcml/debugging/gdbserver.h]: Do we need `x`? `xml` will always be initialized to an empty string, so we can just remove `x`  and use the default constructor for `xml`.

Comment (SimonWin) [reply include/vcml/debugging/gdbserver.h]: This is as a placeholder for later when we maybe want to create the xml file in `add_target`. That might be necessary if we want to use the xml file to differentiate between different architectures of the targets.

**Pull request merged**